compiler/rustc_parse/src/parser/mod.rs

```diff
@@ -595,6 +609,33 @@ impl<'a> Parser<'a> {
         }
     }
 
+    /// Eats a keyword, optionally ignoring the case.
+    /// If the case differs (and is ignored) an error is issued.
+    /// This is useful for recovery.
+    fn eat_keyword_case(&mut self, kw: Symbol, case_insensitive: bool) -> bool {
```

Comment:
What happens if you do the insensitive check unconditionally? I guess that doesn't work because you need to pass in that the next token can be the keyword *or at least* an arbitrary ident... 🤔

Reply:
Yeah, being always case insensitive doesn't work because you may have a `keyword | ident`, for example here:
https://github.com/rust-lang/rust/blob/6a25a433b62701ed48e85857a3b9f29bc4ba758f/compiler/rustc_parse/src/parser/item.rs#L219
If we'd always use case insensitive checks, then `Use!()` would stop working. I had similar problems while working on this, where
```rust
impl Const /* <-- type */ {}
```
was wrongly parsed as `impl const /* waiting for a trait */`